WcaOnRails/app/models/user.rb

```diff
@@ -861,6 +861,20 @@ def notify_of_results_posted(competition)
     end
   end
 
+  def maybe_assign_wca_id_by_results(competition, notify = true)
+    if !wca_id && !unconfirmed_wca_id
+      matches = []
+      unless country.nil?
+        matches = competition.competitors.where(name: name, year: dob.year, month: dob.month, day: dob.day, gender: gender, countryId: country.id).to_a
+      end
+      if matches.size == 1 && matches.first.user.nil?
+        update(wca_id: matches.first.wca_id)
+      elsif notify
```

Comment:
Isn't this always going to be true?

Reply:
Which condition do you mean?

Reply:
Ah, if you're referring to `notify`, it's for two purposes: one is to do a mass update via the ruby console to assign all matching WCA IDs across all comps ever done, and the other is to easily support not sending notifications in the future.